core/src/yamlscript/compiler.clj

```diff
@@ -14,6 +14,8 @@
    [clojure.pprint]
```

Comment:
I would move the logic out of yamlscript/compiler into its own library.
Maybe `yamlscript/compiler/cpp.clj`?